packages/xml-views-validation/src/validators/elements/cardinality-of-aggregation.ts

```diff
@@ -0,0 +1,87 @@
+import { filter, forEach, sortBy, map } from "lodash";
+import { XMLElement } from "@xml-tools/ast";
+import { UI5SemanticModel } from "@ui5-language-assistant/semantic-model-types";
+import { getUI5AggregationByXMLElement } from "@ui5-language-assistant/logic-utils";
+import { CardinalityOfAggregationIssue } from "../../../api";
+
+export function validateCardinality(
+  xmlElement: XMLElement,
+  model: UI5SemanticModel
+): CardinalityOfAggregationIssue[] {
+  const aggregation = getUI5AggregationByXMLElement(xmlElement, model);
+  if (aggregation === undefined || xmlElement.syntax.openName === undefined) {
+    return [];
+  }
+
+  const allCousinsElements = getAllICousinsElements(xmlElement);
+  if (allCousinsElements.length > 1 && aggregation.cardinality === "0..1") {
```

Comment:
I would flip this logic to check if there are **NO** issues and explicitly return `[]`
as it would:
- be consistent with how other empty conditions are handled (line 12-14)
- Would not nest the main logic of this function.
- Make (imho) the flow clearer:
  1. Check X --> No problem.
  2. Check Y --> No problem.
  3. Must Be Problem -> Return None Empty Array of issues